### PR TITLE
Cross browser compatibility

### PR DESCRIPTION
**Pull request by felipegtx**
And a lauch.json file to allow compatibility (to use the IDE to debug the code)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -25,6 +25,9 @@ gulp
 
 When you're ready, continue with next steps:
 
+- Run ```sh
+gulp uglify
```
Comment (fdaciuk) [inline CONTRIBUTING.md]: Actually, you need to run just `gulp`, or `gulp webserver test lint` to run tests and assure that the code has no lint errors.

File: src/ajax.js [hunk 2]
```diff
@@ -40,7 +40,7 @@
     };
 
     $private.XHRConnection = function XHRConnection( type, url, data ) {
-      var xhr = new XMLHttpRequest();
+      var xhr = new(XMLHttpRequest || ActiveXObject)('MSXML2.XMLHTTP.3.0');
```
Comment (fdaciuk) [inline src/ajax.js]: I don't know if it is worth supporting for ie8-.

File: CONTRIBUTING.md (hunk 1 above)
Comment (felipegtx) [inline CONTRIBUTING.md]: This is for the minification. Not for the tests.

Em seg, 20 de jul de 2015 às 19:16, Fernando Daciuk <
notifications@github.com> escreveu:

> In CONTRIBUTING.md
> <https://github.com/fdaciuk/ajax/pull/1#discussion_r35050840>:
>
> > @@ -25,6 +25,9 @@ gulp
> >
> >  When you're ready, continue with next steps:
> >
> > +- Run ```sh
> > +gulp uglify
>
> Actually, you need to run just gulp, or gulp webserver test lint to run
> tests and assure that the code has no lint errors.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fdaciuk/ajax/pull/1/files#r35050840>.
>
-- 
Sent from my mobile

**Comment by fdaciuk** [comment]
I think `.settings/launch.json` file should be included on your `~/.gitignore`. That file is specific for your IDE.

Do you know if IE9+ supports `XMLHttpRequest` object?

**Comment by fdaciuk** [comment]
And thanks for contributing @felipegtx! :smile:

File: CONTRIBUTING.md (hunk 1 above)
Comment (fdaciuk) [inline CONTRIBUTING.md]: Ah, ok xD
I will review this file. This code is broken. :(

**Comment by felipegtx** [comment]
In IE it serves not only as the default XMLHTTP construct but also as a fallback in cases where the native XmlHttp is disabled by the user (given some security policy or something alike). Take a look at: 
https://msdn.microsoft.com/en-us/library/ms537505(v=vs.85).aspx

**Comment by felipegtx** [comment]
Regarding the VSCode file, I can remove it np. When I saw the ```.EditorConfig``` file in your solution I though it would be nice to have something extra targeting yet another Open Source IDE to help developers run the solution with the least possible effort.

File: src/ajax.js (hunk 2 above)
Comment (felipegtx) [inline src/ajax.js]: IMHO it`s not as if this is some crazy hacking, It`s just a fallback that is here to help the lib usage to be as broad as possible.

**Comment by fdaciuk** [comment]
> In IE it (the activeX model) serves not only as the default XMLHTTP construct (up until IE 8, I guess) but also as a fallback in cases where the native XmlHttp is disabled by the user (given some security policy or something alike). Take a look at: 
> https://msdn.microsoft.com/en-us/library/ms537505(v=vs.85).aspx
> 
> Edit: found it. It was on IE7 that the native XMLHttp object was introduced

@felipegtx got it! But do you know if this object supports all methods of the XMLHttpRequest object? That's why in future I will implement other functionalities like upload, CORS, abort requests, etc.

And this code can become huge :)

**Comment by fdaciuk** [comment]
> Regarding the VSCode file, I can remove it np. When I saw the .EditorConfig file in your solution I though it would be nice to have something extra targeting yet another Open Source IDE to help developers run the solution with the least possible effort.

Yep, .editorconfig and .jscs is just for maintain the code patterns. It is not for specific editors :)

File: CONTRIBUTING.md (hunk 1 above)
Comment (fdaciuk) [inline CONTRIBUTING.md]: I fixed this file: [CONTRIBUTING.md](https://github.com/fdaciuk/ajax/blob/master/CONTRIBUTING.md) :smile:

**Comment by felipegtx** [comment]
@fdaciuk The File API is only available for [IE>=10](http://caniuse.com/#feat=fileapi). Prior to that the solution was the [*hidden iframe* hack](https://www.filepicker.com/blog/uncategorized/hacking-a-file-api-onto-ie8/).
For CORS, specifically for IE8 and 9 you need to use the **XDomainRequest**. For details and caveats on the usage, take a look [here](http://blogs.msdn.com/b/ieinternals/archive/2010/05/13/xdomainrequest-restrictions-limitations-and-workarounds.aspx)

File: CONTRIBUTING.md (hunk 1 above)
Comment (felipegtx) [inline CONTRIBUTING.md]: Nice!
Yet, now there's a conflict with this PR :disappointed:

**Comment by fdaciuk** [comment]
@felipegtx Thanks for collaboration :)

I will close this PR, but I will study more about this feature xD

Thanks a lot! :smile: :sparkles:

**Pull request closed without merging**